ui/app/AppLayouts/Chat/components/MessageContextMenu.qml

```diff
@@ -273,4 +273,39 @@ PopupMenu {
         icon.height: 16
         enabled: !isSticker && !emojiOnly
     }
+
+    Separator {
+        visible: deleteMessageAction.enabled
+        height: visible ? Style.current.halfPadding : 0
+    }
+
+    Action {
+        id: deleteMessageAction
+        text: qsTr("Delete message")
+        onTriggered: {
+            if (!appSettings.showDeleteMessageWarning) {
+                return chatsModel.messageView.deleteMessage(messageId)
+            }
+
+            let confirmationDialog = openPopup(genericConfirmationDialog, {
+                                                   title: qsTr("Confirm deleting this message"),
+                                                   confirmationText: qsTr("Are you sure you want to delete this message? Be aware that other clients are not guaranteed to delete the message as well."),
+                                                   height: 260,
+                                                   "checkbox.visible": true,
+                                                   executeConfirm: function () {
+                                                       if (confirmationDialog.checkbox.checked) {
+                                                           appSettings.showDeleteMessageWarning = false
+                                                       }
+
+                                                       confirmationDialog.close()
+                                                       chatsModel.messageView.deleteMessage(messageId)
+                                                   }
+                                               })
+        }
+        icon.source: "../../../img/delete.svg"
+        icon.color: Style.current.danger
+        icon.width: 16
+        icon.height: 16
+        enabled: isCurrentUser
```

Comment:
Probably not all message types can be deleted, but I'm not completely sure. Figma design only shous the delete functionality for text, stickers and images